salt/_states/metalk8s_kubernetes.py

```diff
@@ -472,6 +472,58 @@ def namespace_present(name, **kwargs):
     return ret
 
 
+def namespace_annotation_present(
+        name,
+        annotation_key,
+        annotation_value,
+        **kwargs):
+    '''
+    Ensures that the namespace is patched.
+
+    name
+        The name of the namespace
+
+    annotation_key
+        Key of the annotation
+    annotation_value
+        Value of the annotation
+    '''
+
+    ret = {'name': name,
+           'changes': {},
+           'result': False,
+           'comment': ''}
+
+    annotation = __salt__['metalk8s_kubernetes.show_namespace'](
+        name, **kwargs
+    )
+
+    if annotation_key not in annotation:
+        ret['comment'] = 'The annotation to be added does not exist'
+        if __opts__['test']:
+            ret['result'] = None
+            return ret
+
+        __salt__['metalk8s_kubernetes.namespace_add_annotation'](
+            name, annotation_key, annotation_value, **kwargs)
+    else:
+        if __opts__['test']:
+            ret['result'] = None
+            ret['comment'] = 'The annotation is going to be created/updated'
```

Comment:
Here the annotation already exists so not created at all.

Please really copy/paste the function `node_annotation_present` and just edit the function called and some variables name.